include/mbgl/style/expression/parsing_context.hpp

```diff
@@ -52,6 +52,12 @@ class Scope {
 
 } // namespace detail
 
+enum class TypeAnnotationOption {
```

Comment:
Can we include a comment here describing how these are used? The part that I think is confusing is that there's kind of a fourth option (`nullopt`) which I think means something like "coerce or assert depending on expected type".